src/Voron/Util/DataCopier.cs

```diff
@@ -38,6 +38,11 @@ public void ToStream(byte* ptr, long count, Stream output)
 
         public void ToStream(AbstractPager src, long startPage, long numberOfPages, Stream output)
         {
+            // In case of encryption bundle, we don't want to decrypt the data for backup, 
+            // so let's work directly with the underlying encrypted data (Inner pager).
+            if (src is CryptoPager)
```

Comment:
Bad, leak impl details. Make use of something like the `BatchWriter`, which can already handle this.